packages/block-editor/src/components/inserter/media-tab/media-preview.js

```diff
@@ -180,18 +182,34 @@ export function MediaPreview( { media, onClick, category } ) {
 							if ( isBlobURL( img.url ) ) {
 								return;
 							}
-							onClick( {
-								...clonedBlock,
-								attributes: {
+
+							if ( ! hasInsertedBlock.current ) {
+								// Ensure the block is only inserted once.
+								onClick( {
+									...clonedBlock,
+									attributes: {
+										...clonedBlock.attributes,
+										id: img.id,
+										url: img.url,
+									},
+								} );
+
+								// TODO: Move to a new onSuccess callback as part of #61447.
+								createSuccessNotice(
+									__( 'Image uploaded and inserted.' ),
+									{ type: 'snackbar', id: 'inserter-notice' }
+								);
+
+								hasInsertedBlock.current = true;
```

Comment:
When should we reset this ref? Shouldn't we reset it to "false" when the media prop changes?

Reply:
Ah good point, I forgot about that 😄

Reply:
Removed the ref now.